src/execution/operator/aggregate/physical_streaming_window.cpp

```diff
@@ -189,6 +201,37 @@ class StreamingWindowState : public OperatorState {
 			}
 		}
 
+		void ExecuteLead(ExecutionContext &context, DataChunk &input, DataChunk &delayed, Vector &result) {
+			D_ASSERT(offset < 0);
+			// Input has been set up with the number of rows we CAN produce.
+			const idx_t count = input.size();
+			auto &curr = curr_chunk.data[0];
+			// Copy input[buffered:count] => result[:]
+			idx_t pos = 0;
+			idx_t skipped = 0;
+			if (buffered < count) {
+				curr_chunk.Reset();
+				executor.Execute(input, curr_chunk);
+				VectorOperations::Copy(curr, result, count, buffered, pos);
+				pos += count - buffered;
+			} else {
+				//	We may have to skip the front of the delayed buffer too.
+				skipped = buffered - count;
+			}
+			// Copy delayed[skipped:] => result[pos:]
+			if (skipped < delayed.size()) {
+				curr_chunk.Reset();
+				executor.Execute(delayed, curr_chunk);
+				VectorOperations::Copy(curr, result, delayed.size(), skipped, pos);
```

Comment:
When running the benchmark in debug mode I get an out-of-bounds access here when `pos = 2047` and `delayed.size() = 2`, resulting in a write to index `2049` (one past the vector size). I suspect there might be an off-by-one here?

Reply:
It wasn't a fencepost error, I was just copying more values than were needed. That's why it was passing in release mode. Clamping it fixes the problem.